### PR TITLE
Increasing test coverage for TopoNetX (5)

### PR DESCRIPTION
**Pull request by gurug-dev**
- adding testcases for few missed lines in reportviews, fixed two of the bugs in #277 
- adding testcases for CHG matrix methods

### CODE REVIEW

**Comment by gurug-dev** [comment]
Failing testcases are unrelated to changed code, unknown issue with TestMeshDatasets.

File: toponetx/classes/reportviews.py [hunk 1]
```diff
@@ -415,10 +415,10 @@ def __init__(self, name: str = "") -> None:
...
-        if isinstance(hyperedge, Iterable):
-            hyperedge_ = frozenset(hyperedge)
-        elif isinstance(hyperedge, HyperEdge):
+        if isinstance(hyperedge, HyperEdge):
             hyperedge_ = hyperedge.elements
+        elif isinstance(hyperedge, Iterable):
+            hyperedge_ = frozenset(hyperedge)
         elif isinstance(hyperedge, Hashable) and not isinstance(hyperedge, Iterable):
             hyperedge_ = frozenset([hyperedge])
         return hyperedge_
```
Comment (gurug-dev) [inline toponetx/classes/reportviews.py:424]: Reordered the if-else block.

File: toponetx/classes/reportviews.py [hunk 2]
```diff
@@ -458,20 +458,20 @@ def __contains__(self, e: Collection) -> bool:
...
                         return True
                 return False
-        elif isinstance(e, HyperEdge):
+        elif isinstance(e, Iterable):
             if len(e) == 0:
                 return False
             else:
                 for i in all_ranks:
-                    if frozenset(e.elements) in self.hyperedge_dict[i]:
+                    if frozenset(e) in self.hyperedge_dict[i]:
```
Comment (gurug-dev) [inline toponetx/classes/reportviews.py:474]: same as above, reordered the if-else block.

**Comment by mhajij** [comment]
@gurug-dev  all unit tests must pass

**Comment by gurug-dev** [comment]
> @gurug-dev all unit tests must pass

FAILED test/datasets/test_mesh.py::TestMeshDatasets::test_shrec_16 - zipfile.BadZipFile: File is not a zip file
FAILED test/datasets/test_mesh.py::TestMeshDatasets::test_coseg - zipfile.BadZipFile: File is not a zip file

Hi Professor, failling test cases are unrelated to changed code.

**Comment by ninamiolane** [comment]
@gurug-dev why are tests failing? How can we solve it? We can't merge a failing PR.

**Review by ninamiolane — CHANGES_REQUESTED** [review]
We can't merge a PR that fails the test: it will fail the whole repository. Need to fix the fails first.

**Comment by gurug-dev** [comment]
> @gurug-dev why are tests failing? How can we solve it? We can't merge a failing PR.

@ninamiolane - i'm trying to figure it out, the test suite is passes locally, but the tests are failing in my forked branches (without the changes) on github as well. my best guess is some github actions related issue. will raise a PR if i'm able to solve it.

**Comment by gurug-dev** [comment]
@ninamiolane @mhajij - the mesh datasets issue has been fixed, the test suite is now passing. For avoiding further issues, the mesh datasets need to be potentially moved to a server that is not github.

**Review by mhajij — APPROVED** [review]

**Review by mhajij — APPROVED** [review]

**Review by mhajij — APPROVED** [review]

**Review by mhajij — APPROVED** [review]

**Pull request merged**